backend/ohq/views.py

```diff
@@ -484,6 +471,23 @@ def get_queryset(self):
             )
         return prefetch(qs, self.serializer_class)
 
+    @action(detail=False)
+    def staff_active(self, request, course_pk):
+        """
+        Get's the active staff in a course
+        """
+
+        time_threshold = timezone.now() - timedelta(minutes=1)
+
+        staff_active = Membership.objects.filter(
+            Q(course=course_pk)
+            & ~Q(kind=Membership.KIND_STUDENT)
+            & Q(last_active__gt=time_threshold)
+        )
+
+        serializer = self.get_serializer(staff_active, many=True)
+        return Response(serializer.data)
```

Comment:
This definitely works (and is arguably more clear than what I'm about to suggest) but I think you could also have this method just `pass` and then in `get_queryset` check if the action is staff_active and only return active TAs that way

Reply:
Actually, after @eyingxuan's presentation on WS I think we should do it this way so the action is WS subscribable

Reply:
I think DRL doesn't allow us to do WS subscribes to custom actions at the moment so I instead used a query param to see if we want to get staff_active vs. list all staff for a course

Reply:
Hmm, that's unfortunate. @davish is that the case?

Reply:
Yep. You subscribe to a queryset as it's visible in a viewset's get_queryset method with the serializer defined in get_serializer. There's no real concept of actions beyond list and retrieve, and there's no way for DRL to know what the action does and to trigger a broadcast message.